src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,50 @@
+class Quantity(float):
+    """This class instantiates physical quantities using their real values and
+    units.
+
+    Attributes
+    ----------
+    value: Real value
+        Value of quantity is stored as float.
+
+    unit: Unit string
+        Unit of quantity is stored as string.
+
+    Methods
+    -------
+    to(to_unit)
+        Converts to given unit string.
+
+    toSystem(baseSystem)
+        Converts to given base system.
+
+    getDimensions(unit)
+        Extracts dimensions from unit.
+
+    isDimensionless()
+        Determines type of quantity.
+
+    checkDimensionsMatch(quantity1, quantity2)
+        Compares dimensions of 2 quantities.
+
+    Returns
+    -------
+    Quantity instance.
+
+    All the instances of this class are converted to base SI units system to have
+    consistency in all arithmetic operations.
+    """
+
+    def __new__(self, real_value, units_string):
```

Comment:
Is this usual / required?

Reply:
Yes, because without this method we are not able to inherit float class properly.

Reply:
OK, I think you should rename `self` as `cls` though, as that is what it is, right?

Reply:
Done. Thank you.